internal/clients/tenant.go

```diff
@@ -128,3 +136,32 @@ func (t Tenant) GetNamespaceByType(ti TenantInfo, typ string) (r Namespace, err
 	err = fmt.Errorf("could not find tenant %s Jenkins namespace", ti.Data.Attributes.Email)
 	return
 }
+
+// GetNamespace gets namespace given appropriate accessToken
+func (t Tenant) GetNamespace(accessToken string) (*Namespace, error) {
```

Comment:
why *Namespace when GetNamespaceByType returns value? 
IMHO, it should be `func (t Tenant) GetNamespace(accessToken string) (Namespace, error)` and `return namespace, nil`

Reply:
`*Namespace, error` allows to return `nil, err`

Reply:
yes, so what does that save the caller? 
now you have to check the `error` and then check if the object is nil? 
heard of https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare ? https://en.wikipedia.org/wiki/Tony_Hoare

Reply:
Work for the caller is same in both cases.

Reply:
It isn't .. for a pointer, they have to check for nil every single time it is accessed. 
```go
package main

import "fmt"

type person struct {
	name string
}

func value() person {
	return person{}
}

func pointer() *person {
	return nil
}

func main() {

	v := value()
	fmt.Println(v.name)

	p := pointer()
	fmt.Println(p.name)  // ### crash ###
}
```

Reply:
May be a confirmation bias still - https://www.reddit.com/r/golang/comments/2xmnvs/returning_nil_for_a_struct/cp1h15d